aclib/__init__.py

```diff
@@ -34,18 +34,28 @@ def get_dbpath():
 
 
 def get_url(db_name=None, schema_name=None):
-    credential_filename = os.getenv("ACLIB_CREDENTIAL_FILENAME", os.path.expanduser("~") + "/.odbc.ini")
+    username_filename = os.getenv("ACLIB_USERNAME_FILENAME", os.path.expanduser("~") + "/secrets/username")
+    snowflake_access_token_filename = os.getenv("ACLIB_SNOWFLAKE_ACCESSS_TOKEN_FILENAME", os.path.expanduser("~") + "/secrets/snowflake_access_token")
     credential_section = os.getenv("ACLIB_CREDENTIAL_SECTION", "nuvolos")
 
     # Create engine with credentials
     cred = ConfigParser(interpolation=None)
-    if not os.path.exists(credential_filename):
-        raise FileNotFoundError(f"Credentials file {credential_filename} not found")
-    cred.read(credential_filename)
-    credd = dict(cred.items(credential_section))
-    snowflake_host = credd.get('host', "alphacruncher.eu-central-1")
-    url = 'snowflake://' + credd['uid'] + ':' + credd['pwd'] + '@' + snowflake_host + '/?warehouse=' + credd['uid']
-    masked_url = 'snowflake://' + credd['uid'] + ':********' + '@' + snowflake_host + '/?warehouse=' + credd['uid']
+    if not os.path.exists(username_filename) and not os.path.exists(snowflake_access_token_filename):
+        credential_filename = os.getenv("ACLIB_CREDENTIAL_FILENAME", os.path.expanduser("~") + "/.odbc.ini")
+        if not os.path.exists(credential_filename):
+            raise FileNotFoundError(f"Credentials file {credential_filename} not found")
+        cred.read(credential_filename)
+        credd = dict(cred.items(credential_section))
+        snowflake_host = credd.get('host', "alphacruncher.eu-central-1")
+        url = 'snowflake://' + credd['uid'] + ':' + credd['pwd'] + '@' + snowflake_host + '/?warehouse=' + credd['uid']
+        masked_url = 'snowflake://' + credd['uid'] + ':********' + '@' + snowflake_host + '/?warehouse=' + credd['uid']
+    else:
+        cred.read([username_filename, snowflake_access_token_filename])
+        credd = dict(cred.items(credential_section))
+        snowflake_host = credd.get('host', "alphacruncher.eu-central-1")
```

Comment:
In this case, let's use the os.getenv("ACLIB_SNOWFLAKE_HOST", "alphacruncher.eu-central-1") pattern instead.